n3fit/src/n3fit/layers/observable.py

```diff
@@ -42,36 +43,104 @@ def __init__(self, fktable_data, fktable_arr, operation_name, nfl=14, **kwargs):
         super(MetaLayer, self).__init__(**kwargs)
 
         self.nfl = nfl
+        self.num_replicas = None  # set in build
+        self.compute_observable = None  # set in build
```

Comment:
Add a comment about `compute_observable` being a function with signature (pdf, masked_pdf) that need to be overwritten by any children of this class.

Actually, maybe it makes sense to make `compute_observable` be an abstract method and then in `DIS.py` and `DY.py` the choice of which (outside) function to use becomes

```python
def compute_observable(self, pdf, fk):
    if self._one_replica:
        return _compute_dis_observable_one_replica(pdf, fk)
    return _compute_dis_observable_many_replica(pdf, fk)

Reply:
I added the comment, wanted to avoid these if statements in the call. It probably doesn't matter but I thought it looked cleaner. But can make it an abstract method if you prefer.